api/mod/tests/conftest.py

```diff
@@ -30,3 +41,47 @@ def session() -> Generator:
     yield SETTINGS
 
     shutil.rmtree(SETTINGS.data_folder)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def apply_migrations() -> Generator:
+    """Apply migrations to test database"""
+    config = Config(str(Path.cwd().joinpath("alembic.ini")))
+    config.set_main_option(
+        "script_location", str(Path.cwd().joinpath("mod").joinpath("alembic"))
+    )
+    config.set_main_option("sqlalchemy.url", SETTINGS.database_url)
+    alembic.command.upgrade(config, "head")
+    yield
+    alembic.command.downgrade(config, "base")
+
+
+@pytest.fixture(autouse=True)
+def db() -> scoped_session:
+    """
+    Provides a database session that is scoped to the test function
+    and is automatically rolled back after the test
+    """
+    connection = engine.connect()
+    transaction = connection.begin()
+    session = scoped_session(
+        sessionmaker(autocommit=False, autoflush=False, bind=connection)
+    )
+
+    def override_get_db() -> scoped_session:
+        return session
+
+    APP.dependency_overrides[get_db] = override_get_db
+    yield session
+    transaction.rollback()
+    connection.close()
+
+
+@pytest.fixture
+def user() -> UserCreate:
+    """Creates a basic Annotation"""
+    return UserCreate(
+        username="AnnotoUser#1337",
+        password="password",
+        email="annoto@team-folivora.com",
+    )
```

Comment:
I think this really does not belong into our global conftest

Reply:
It's especially confusing in the tests (e.g. in `test_login_router.py`), where you use the fixture from here with the type annotation from `mod.src.models.user`

Reply:
Ok I see the point with the global conftest, but as you pointed out, it is used in two router tests. We could make a conftest for the integration tests folder, would that be better?
I dont think that solves the import thing you mentioned, but I also do not really understand what you mean there 😅